src/main/java/net/minecraftforge/common/ForgeHooks.java

```diff
@@ -1012,6 +1012,22 @@ public static void onEmptyLeftClick(EntityPlayer player, ItemStack stack)
         MinecraftForge.EVENT_BUS.post(new PlayerInteractEvent.LeftClickEmpty(player, stack));
     }
 
+    public static float getCriticalHitModifier(EntityPlayer player, Entity target, boolean vanillaCritical)
+    {
+        net.minecraftforge.event.entity.player.CriticalHitEvent.Decision hitResult = new net.minecraftforge.event.entity.player.CriticalHitEvent.Decision(player, target);
+        net.minecraftforge.common.MinecraftForge.EVENT_BUS.post(hitResult);
+        if(hitResult.getResult() != Event.Result.DENY)
+        {
+            if(hitResult.getResult() == Event.Result.ALLOW || vanillaCritical)
+            {
+                net.minecraftforge.event.entity.player.CriticalHitEvent.HitModifier hit = new net.minecraftforge.event.entity.player.CriticalHitEvent.HitModifier(player, target);
+                net.minecraftforge.common.MinecraftForge.EVENT_BUS.post(hit);
+                return hit.getDamageModifier();
+            }
+        }
+        return 1F;
+    }
+    
```

Comment:
this is better in that it's not in the patch anymore, but it's still significantly more complicated than what I commented earlier.